app/ports/models/buildhistory.py

```diff
@@ -103,6 +108,8 @@ def load_database(data):
 
             for build_number in range(build_in_database, last_build_number):
                 build_data = get_data_from_url(get_url_json(buildername, build_number))
+                if not build_data:
+                    continue
```

Comment:
Reading this code I suspect that if fetching just one build out of ten would fail, we'll happily continue to the next number, but we will never recover later (never attempt to fetch that one build again) and we'll always keep the hole in the data.

Please correct me if I'm wrong.

Reply:
Oh, that's right, thank you. I believe a `break` would be a better choice, because the app is very well capable of recovering from the last state.

Reply:
I have made the changes, and now the flow is: 

1. We loop over all the builders.
2. For each builder, we try to fetch the most recent build, if the process fails, we continue to next builder.
3. Then for each builder, we obtain the highest build id from the database. And loop from 'highest in database' to 'latest build id from buildbot'.
4. If anywhere in between, the process fails- it breaks and moves on to next builder.